pulsar-client-api/src/main/java/org/apache/pulsar/client/api/ProducerBuilder.java

```diff
@@ -349,6 +349,20 @@
      */
     ProducerBuilder<T> batchingMaxPublishDelay(long batchDelay, TimeUnit timeUnit);
 
+    /**
+     * Set the partition switch frequency while batching of messages is enabled and
+     * using round-robin routing mode for non-keyed message <i>default: 10</i>.
+     *
+     * The time period of partition switch is frequency * batchingMaxPublishDelay. During this period,
+     * all messages arrives will be route to the same partition.
+     *
+     * @param frequency the frequency of partition switch
+     * @return the producer builder instance
+     * @see #messageRoutingMode(MessageRoutingMode)
+     * @see #batchingMaxPublishDelay(long, TimeUnit)
+     */
+    ProducerBuilder<T> batchingPartitionSwitchFrequency(int frequency);
```

Comment:
It is better to add both time interval and time unit in the method, by following the convention of other time related settings.

Reply:
The current frequency is used to compute switch delay in `frequency` * `maxBatchDelayMicros`. Do you mean I should make it a standalone variable? Then if users didn't set this variable, or the user is setting a variable less than `maxBatchDelay`, I need to use `maxBatchDelay` in round-robin partitioner?

Reply:
Oh I misread the code. I see it now.  Then that's fine to me.